ui/app/settings/page.tsx

```diff
@@ -134,12 +178,16 @@ const SettingsPage = () => {
     fetchSettings();
   }, []);
 
-  const totalPages = Math.ceil(settings.length / ROWS_PER_PAGE);
-
-  const displayedSettings = useMemo(() => {
-    const filteredSettings = settings.filter((setting) =>
+  const filteredSettings = useMemo(() => {
+    return settings.filter((setting) =>
       setting.config_name.toLowerCase().includes(searchQuery.toLowerCase())
     );
+  }, [settings, searchQuery]);
+  const totalPages = useMemo(
+    () => Math.ceil(filteredSettings.length / ROWS_PER_PAGE),
+    [filteredSettings]
+  );
```

Comment:
```suggestion
  const totalPages = Math.ceil(filteredSettings.length / ROWS_PER_PAGE);
```
not worth overhead of hook

Reply:
won't update with search term then

Reply:
yes it will, useMemo is only useful to prevent reexecution